subtests/docker_cli/systemd/systemd.py

```diff
@@ -65,22 +52,16 @@ def sysd_action(action=None, sysd_unit=None):
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def init_unitfile(self, separator=None, changeto=None):
```

Comment:
Agree with @cevich. One addition: `separator` is misnamed. And I think the function is too. And I'm not sure I like the multiple invocations and copies. I think there's got to be a cleaner way, I'm just not sure what that is. Maybe:

    def write_unitfile(self, env):
         ...open original (git-controlled) file for reading
         ...open output file (/etc/systemd) for writing
          ...while read:
               if line.startswith('Environment='):
                     extract the right-hand side, split on the '=' sign
                     if left-hand side in env dict:
                         line = 'Environment={}={}'.format(left-hand-side, env[right-hand-side])
                     write line to output file